File: backend-v2/app/api/routes/feedback.py
from fastapi import APIRouter

from app.schemas.feedback import FeedbackCreate, FeedbackRead, FeedbackUpdate

router = APIRouter()


@router.post("", response_model=FeedbackRead)
def create_feedback(payload: FeedbackCreate):
    return FeedbackRead(
        id="feedback-placeholder",
        record_id=payload.record_id,
        judgement=payload.judgement,
        corrected_label=payload.corrected_label,
        comment=payload.comment,
        reviewer=payload.reviewer,
    )


@router.patch("/{feedback_id}", response_model=FeedbackRead)
def update_feedback(feedback_id: str, payload: FeedbackUpdate):
    return FeedbackRead(
        id=feedback_id,
        record_id=payload.record_id or "record-placeholder",
        judgement=payload.judgement or "correct",
        corrected_label=payload.corrected_label,
        comment=payload.comment,
        reviewer=payload.reviewer or "reviewer",
    )

File: frontend-v2/src/shared/api/feedback.ts
// @prd /prd/modules/feedback-center.md §5 数据模型 - PredictionFeedback
import { apiClient } from './client';

export type Feedback = {
  id: string;
  record_id: string;
  judgement: string;
  corrected_label: string | null;
  comment: string | null;
  reviewer: string;
  created_at: string | null;
};

export async function listFeedback(params?: { recordId?: string }) {
  const response = await apiClient.get<Feedback[]>('/api/feedback', {
    params: {
      record_id: params?.recordId || undefined,
    },
  });
  return response.data;
}

export async function createFeedback(payload: {
  recordId: string;
  judgement: string;
  correctedLabel?: string;
  comment?: string;
}) {
  const response = await apiClient.post<Feedback>('/api/feedback', {
    record_id: payload.recordId,
    judgement: payload.judgement,
    corrected_label: payload.correctedLabel || null,
    comment: payload.comment || null,
  });
  return response.data;
}

export async function updateFeedback(
  feedbackId: string,
  payload: {
    judgement?: string;
    correctedLabel?: string;
    comment?: string;
  },
) {
  const response = await apiClient.patch<Feedback>(`/api/feedback/${feedbackId}`, {
    judgement: payload.judgement,
    corrected_label: payload.correctedLabel || null,
    comment: payload.comment || null,
  });
  return response.data;
}
